### PR TITLE
docs: translate hooks API reference

### PR DESCRIPTION
**Pull request by ivan-ha**
Done.
~Not finished all, create this draft PR to kickstart review first.~

TODO:
- [x] `useImperativeHandle`
- [x] `useLayoutEffect`
- [x] `useDebugValue`
- [x] sidebar menu

### CODE REVIEW

**Comment by ivan-ha** [comment]
@neighborhood999 the PR is ready for review, thanks!

**Comment by neighborhood999** [comment]
Hi @ivan-ha 

the hooks-reference is conflict.

**Comment by ivan-ha** [comment]
thanks @neighborhood999 , resolved

File: content/docs/hooks-reference.md [hunk 1]
```diff
@@ -297,11 +297,11 @@ function Counter({initialCount}) {
 }
 ```
 
-#### Bailing out of a dispatch {#bailing-out-of-a-dispatch}
+#### 跳過 dispatch {#bailing-out-of-a-dispatch}
 
-If you return the same value from a Reducer Hook as the current state, React will bail out without rendering the children or firing effects. (React uses the [`Object.is` comparison algorithm](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/is#Description).)
+如果你在 reducer hook 返回的值與當前的 state 相同，React 將會跳過子組件的 render 及 effect 的執行。（React 使用 [`Object.is` 比較算法](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/is#Description)。）
```
Comment (neighborhood999) [inline content/docs/hooks-reference.md]:
```suggestion
如果你在 Reducer Hook 回傳的值與目前的 state 相同，React 將會跳過 child component 的 render 及 effect 的執行。（React 使用 [`Object.is` 比較演算法](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/is#Description)。）
```

File: content/docs/hooks-reference.md [hunk 2]
```diff
@@ -297,11 +297,11 @@ function Counter({initialCount}) {
...
 
-Note that React may still need to render that specific component again before bailing out. That shouldn't be a concern because React won't unnecessarily go "deeper" into the tree. If you're doing expensive calculations while rendering, you can optimize them with `useMemo`.
+請注意 React 可能仍需要在跳過 render 之前 render 該 component。這不應該是個問題，因為 React 不會不必要地「深入」到組件樹中。如果你在 render 當中執行了高昂的計算，你可以使用 `useMemo` 來優化。
```
Comment (neighborhood999) [inline content/docs/hooks-reference.md]:
```suggestion
請注意 React 可能仍需要在跳過 render 之前 render 該 component。這不應該是個問題，因為 React 不會不必要地「深入」到 tree 中。如果你在 render 當中執行了昂貴的計算，你可以使用 `useMemo` 來最佳化。
```

File: content/docs/hooks-reference.md [hunk 3]
```diff
@@ -314,49 +314,49 @@ const memoizedCallback = useCallback(
 );
 ```
 
-Returns a [memoized](https://en.wikipedia.org/wiki/Memoization) callback.
+返回一個 [memoized](https://en.wikipedia.org/wiki/Memoization) 的 callback。
```
Comment (neighborhood999) [inline content/docs/hooks-reference.md]:
```suggestion
回傳一個 [memoized](https://en.wikipedia.org/wiki/Memoization) 的 callback。
```

File: content/docs/hooks-reference.md [hunk 4]
```diff
@@ -314,49 +314,49 @@ const memoizedCallback = useCallback(
...
 
-Pass an inline callback and an array of dependencies. `useCallback` will return a memoized version of the callback that only changes if one of the dependencies has changed. This is useful when passing callbacks to optimized child components that rely on reference equality to prevent unnecessary renders (e.g. `shouldComponentUpdate`).
+傳遞一個內聯 callback 及依賴 array。`useCallback` 會返回該 callback 的 memoized 版本，它僅在依賴改變時才會更新。當傳遞 callback 到已優化使用引用相等性來避免不必要 render （例如 `shouldComponentUpdate`）的子 component 時，它將非常有用。
```
Comment (neighborhood999) [inline content/docs/hooks-reference.md]:
```suggestion
傳遞一個 inline callback 及依賴 array。`useCallback` 會回傳該 callback 的 memoized 版本，它僅在依賴改變時才會更新。當傳遞 callback 到已經最佳化的 child component 時非常有用，這些 child component 依賴於引用相等性來防止不必要的 render（例如，`shouldComponentUpdate`）
```

File: content/docs/hooks-reference.md [hunk 5]
```diff
@@ -314,49 +314,49 @@ const memoizedCallback = useCallback(
...
 
-`useCallback(fn, deps)` is equivalent to `useMemo(() => fn, deps)`.
+`useCallback(fn, deps)` 相等於 `useMemo(() => fn, deps)`。
 
-> Note
+> 注意
 >
-> The array of dependencies is not passed as arguments to the callback. Conceptually, though, that's what they represent: every value referenced inside the callback should also appear in the dependencies array. In the future, a sufficiently advanced compiler could create this array automatically.
+> 依賴 array 並不作為傳到 callback 的參數。但從概念上來説，這是它所代表的：所有在 callback 中引用的值都應該出現在依賴 array 中。在未來，一個足夠先進的編譯器可以自動創建這 array。
```
Comment (neighborhood999) [inline content/docs/hooks-reference.md]:
```suggestion
> 依賴 array 並不作為傳到 callback 的參數。但從概念上來説，這是它所代表的：所有在 callback 中引用的值都應該出現在依賴 array 中。在未來，一個足夠先進的編譯器可以自動建立這個 array。
```

File: content/docs/hooks-reference.md [hunk 6]
```diff
@@ -314,49 +314,49 @@ const memoizedCallback = useCallback(
...
+> 我們建議使用 [`eslint-plugin-react-hooks`](https://www.npmjs.com/package/eslint-plugin-react-hooks#installation) 中的 [`exhaustive-deps`](https://github.com/facebook/react/issues/14920) 規則。它會在依賴錯誤時發出警告並提出修正建議。
 
 ### `useMemo` {#usememo}
 
 ```js
 const memoizedValue = useMemo(() => computeExpensiveValue(a, b), [a, b]);
 ```
 
-Returns a [memoized](https://en.wikipedia.org/wiki/Memoization) value.
+返回一個 [memoized](https://en.wikipedia.org/wiki/Memoization) 的值。
```
Comment (neighborhood999) [inline content/docs/hooks-reference.md]:
```suggestion
回傳一個 [memoized](https://en.wikipedia.org/wiki/Memoization) 的值。
```

File: content/docs/hooks-reference.md [hunk 7]
```diff
@@ -314,49 +314,49 @@ const memoizedCallback = useCallback(
...
 
-Pass a "create" function and an array of dependencies. `useMemo` will only recompute the memoized value when one of the dependencies has changed. This optimization helps to avoid expensive calculations on every render.
+傳遞一個「創建」function 及依賴 array。`useMemo` 只會在依賴改變時才重新計算 memoized 的值。這個優化可以避免在每次 render 都進行昂貴的計算。
```
Comment (neighborhood999) [inline content/docs/hooks-reference.md]:
```suggestion
傳遞一個「建立」function 及依賴 array。`useMemo` 只會在依賴改變時才重新計算 memoized 的值。這個最佳化可以避免在每次 render 都進行昂貴的計算。
```

File: content/docs/hooks-reference.md [hunk 8]
```diff
@@ -314,49 +314,49 @@ const memoizedCallback = useCallback(
...
 
-Remember that the function passed to `useMemo` runs during rendering. Don't do anything there that you wouldn't normally do while rendering. For example, side effects belong in `useEffect`, not `useMemo`.
+要謹記傳到 `useMemo` 的 function 會在 render 期間執行。不要做一些通常不會在 render 期間做的事情。例如，處理副作用屬於 `useEffect`，而不是 `useMemo`。
```
Comment (neighborhood999) [inline content/docs/hooks-reference.md]:
```suggestion
要謹記傳到 `useMemo` 的 function 會在 render 期間執行。不要做一些通常不會在 render 期間做的事情。例如，處理 side effect 屬於 `useEffect`，而不是 `useMemo`。
```

File: content/docs/hooks-reference.md [hunk 9]
```diff
@@ -314,49 +314,49 @@ const memoizedCallback = useCallback(
...
 
-If no array is provided, a new value will be computed on every render.
+如果沒有提供 array，每次 render 時都會計算新的值。
 
-**You may rely on `useMemo` as a performance optimization, not as a semantic guarantee.** In the future, React may choose to "forget" some previously memoized values and recalculate them on next render, e.g. to free memory for offscreen components. Write your code so that it still works without `useMemo` — and then add it to optimize performance.
+**你可以把 `useMemo` 作為性能優化的手段，但請不要把它當作成語意上的保證。**在將來，React 可能會選擇「忘記」某些之前已 memorized 的值並在下一次 render 時重新計算，例如為已離開屏幕的 component 釋放內存。先編寫沒有 `useMemo` 也可執行的代碼 — 然後再加入它來優化效能。
```
Comment (neighborhood999) [inline content/docs/hooks-reference.md]:
```suggestion
**你可以把 `useMemo` 作為效能最佳化的手段，但請不要把它當作成語意上的保證。**在將來，React 可能會選擇「忘記」某些之前已 memorize 的值並在下一次 render 時重新計算，例如，為已離開螢幕的 component 釋放記憶體。先撰寫沒有 `useMemo` 也可執行的代碼 — 然後再加入它來做效能最佳化。
```

File: content/docs/hooks-reference.md [hunk 10]
```diff
@@ -314,49 +314,49 @@ const memoizedCallback = useCallback(
...
 
-> Note
+> 注意
 >
-> The array of dependencies is not passed as arguments to the function. Conceptually, though, that's what they represent: every value referenced inside the function should also appear in the dependencies array. In the future, a sufficiently advanced compiler could create this array automatically.
+> 依賴 array 並不作為傳到 function 的參數。但從概念上來説，這是它所代表的：所有在 callback 中引用的值都應該出現在依賴 array 中。在未來，一個足夠先進的編譯器可以自動創建這 array。
```
Comment (neighborhood999) [inline content/docs/hooks-reference.md]:
```suggestion
> 依賴 array 並不作為傳到 function 的參數。但從概念上來説，這是它所代表的：所有在 callback 中引用的值都應該出現在依賴 array 中。在未來，一個足夠先進的編譯器可以自動建立這個 array。
```

File: content/docs/hooks-reference.md [hunk 11]
```diff
@@ -314,49 +314,49 @@ const memoizedCallback = useCallback(
...
+> 我們建議使用 [`eslint-plugin-react-hooks`](https://www.npmjs.com/package/eslint-plugin-react-hooks#installation) 中的 [`exhaustive-deps`](https://github.com/facebook/react/issues/14920) 規則。它會在依賴錯誤時發出警告並提出修正建議。
 
 ### `useRef` {#useref}
 
 ```js
 const refContainer = useRef(initialValue);
 ```
 
-`useRef` returns a mutable ref object whose `.current` property is initialized to the passed argument (`initialValue`). The returned object will persist for the full lifetime of the component.
+`useRef` 返回一個可變異的 ref object，其 `.current` 屬性被初始為傳入的參數（`initialValue`）。返回的 object 在 component 的生命週期將保持不變。
```
Comment (neighborhood999) [inline content/docs/hooks-reference.md]:
```suggestion
`useRef` 回傳一個 mutable 的 ref object，`.current` 屬性被初始為傳入的參數（`initialValue`）。回傳的 object 在 component 的生命週期將保持不變。
```

File: content/docs/hooks-reference.md [hunk 12]
```diff
@@ -314,49 +314,49 @@ const memoizedCallback = useCallback(
...
 
-A common use case is to access a child imperatively:
+一個常見的用例就是命令式地訪問子元件：
```
Comment (neighborhood999) [inline content/docs/hooks-reference.md]:
```suggestion
一個常見的使用情境就是命令式的訪問 child component：
```

File: content/docs/hooks-reference.md [hunk 13]
```diff
@@ -374,15 +374,15 @@ function TextInputWithFocusButton() {
 }
 ```
 
-Essentially, `useRef` is like a "box" that can hold a mutable value in its `.current` property.
+本質上，`useRef` 就像一個保存可變異值於其 `.current` 屬性的「盒子」。
```
Comment (neighborhood999) [inline content/docs/hooks-reference.md]:
```suggestion
本質上，`useRef` 就像一個可以持有 mutable 的值在 `.current` 屬性的「盒子」。
```

File: content/docs/hooks-reference.md [hunk 14]
```diff
@@ -374,15 +374,15 @@ function TextInputWithFocusButton() {
...
 
-You might be familiar with refs primarily as a way to [access the DOM](/docs/refs-and-the-dom.html). If you pass a ref object to React with `<div ref={myRef} />`, React will set its `.current` property to the corresponding DOM node whenever that node changes.
+你應該熟悉 ref 一種主要是用來[訪問 DOM](/docs/refs-and-the-dom.html)的方式。如果你在 React 中以 `<div ref={myRef} />` 傳入 ref object，無論節點如何改變，React 都會將其 `.current` 屬性設為相應的 DOM 節點。
```
Comment (neighborhood999) [inline content/docs/hooks-reference.md]:
```suggestion
你應該熟悉 ref，它一種主要是用來[訪問 DOM](/docs/refs-and-the-dom.html)的方式。如果你在 React 中以 `<div ref={myRef} />` 傳入 ref object，無論節點如何改變，React 都會將其 `.current` 屬性設為相應的 DOM 節點。
```

File: content/docs/hooks-reference.md [hunk 15]
```diff
@@ -374,15 +374,15 @@ function TextInputWithFocusButton() {
...
 
-However, `useRef()` is useful for more than the `ref` attribute. It's [handy for keeping any mutable value around](/docs/hooks-faq.html#is-there-something-like-instance-variables) similar to how you'd use instance fields in classes.
+然而，`useRef()` 比 `ref` 屬性更有用。它可以[很方便地保存任何可變異的值](/docs/hooks-faq.html#is-there-something-like-instance-variables)，跟 class 中的實例字段類似。
```
Comment (neighborhood999) [inline content/docs/hooks-reference.md]:
```suggestion
然而，`useRef()` 比 `ref` 屬性更有用。它可以[很方便地持有任何 mutable 的值](/docs/hooks-faq.html#is-there-something-like-instance-variables)，跟 class 中的 instance field 類似。
```

File: content/docs/hooks-reference.md [hunk 16]
```diff
@@ -374,15 +374,15 @@ function TextInputWithFocusButton() {
...
 
-This works because `useRef()` creates a plain JavaScript object. The only difference between `useRef()` and creating a `{current: ...}` object yourself is that `useRef` will give you the same ref object on every render.
+這是因為 `useRef()` 會創建一個普通的 JavaScript object。`useRef()` 和自建一個 `{current: ...}` object 的唯一不同是 `useRef` 在每次 render 時都會給你同一個的 ref object。
```
Comment (neighborhood999) [inline content/docs/hooks-reference.md]:
```suggestion
這是因為 `useRef()` 會建立一個普通的 JavaScript object。`useRef()` 和自建一個 `{current: ...}` object 的唯一不同是，`useRef` 在每次 render 時都會給你同一個的 ref object。
```

File: content/docs/hooks-reference.md [hunk 17]
```diff
@@ -374,15 +374,15 @@ function TextInputWithFocusButton() {
...
 
-Keep in mind that `useRef` *doesn't* notify you when its content changes. Mutating the `.current` property doesn't cause a re-render. If you want to run some code when React attaches or detaches a ref to a DOM node, you may want to use a [callback ref](/docs/hooks-faq.html#how-can-i-measure-a-dom-node) instead.
+請記住 `useRef` 在其內容有變化時並*不會*通知你。變更 `.current` 屬性不會觸發重新 render。如果你想要在 React 綁定或解綁 DOM 節點的 ref 時執行代碼，你可能需要使用 [callback ref](/docs/hooks-faq.html#how-can-i-measure-a-dom-node) 來實現。
```
Comment (neighborhood999) [inline content/docs/hooks-reference.md]:
```suggestion
請記住 `useRef` 在其內容有變化時並*不會*通知你。變更 `.current` 屬性不會觸發重新 render。如果你想要在 React 綁定或解綁 DOM 節點的 ref 時執行程式碼，你可能需要使用 [callback ref](/docs/hooks-faq.html#how-can-i-measure-a-dom-node) 來實現。
```

File: content/docs/hooks-reference.md [hunk 18]
```diff
@@ -391,7 +391,7 @@ Keep in mind that `useRef` *doesn't* notify you when its content changes. Mutati
 useImperativeHandle(ref, createHandle, [deps])
 ```
 
-`useImperativeHandle` customizes the instance value that is exposed to parent components when using `ref`. As always, imperative code using refs should be avoided in most cases. `useImperativeHandle` should be used with `forwardRef`:
+`useImperativeHandle` 可以讓使用 `ref` 時能向父 component 暴露自訂義的實例值。一如既往，在大多數的情況下應避免使用 ref 的命令式代碼。`useImperativeHandle` 應與 `forwardRef`  一同使用：
```
Comment (neighborhood999) [inline content/docs/hooks-reference.md]:
```suggestion
`useImperativeHandle` 可以讓使用 `ref` 時能向父 component 暴露自定義的 instance 值。一如既往，在大多數的情況下應避免使用 ref 的命令式代碼。`useImperativeHandle` 應與 `forwardRef` 一同使用：
```

File: content/docs/hooks-reference.md [hunk 19]
```diff
@@ -406,31 +406,31 @@ function FancyInput(props, ref) {
 FancyInput = forwardRef(FancyInput);
 ```
 
-In this example, a parent component that renders `<FancyInput ref={fancyInputRef} />` would be able to call `fancyInputRef.current.focus()`.
+在這個例子，render `<FancyInput ref={fancyInputRef} />` 的父 component 能調用 `fancyInputRef.current.focus()`。
```
Comment (neighborhood999) [inline content/docs/hooks-reference.md]:
```suggestion
在這個範例中，render `<FancyInput ref={fancyInputRef} />` 的父 component 能呼叫 `fancyInputRef.current.focus()`。
```

File: content/docs/hooks-reference.md [hunk 20]
```diff
@@ -406,31 +406,31 @@ function FancyInput(props, ref) {
...
 
 ### `useLayoutEffect` {#uselayouteffect}
 
-The signature is identical to `useEffect`, but it fires synchronously after all DOM mutations. Use this to read layout from the DOM and synchronously re-render. Updates scheduled inside `useLayoutEffect` will be flushed synchronously, before the browser has a chance to paint.
+其簽名與 `useEffect` 相同，但它會在所有 DOM 變異後同步調用。使用它來讀取 DOM 佈局並同步重新 render。在瀏覽器執行繪製之前，`useLayoutEffect` 內部的更新計劃將被同步刷新。
```
Comment (neighborhood999) [inline content/docs/hooks-reference.md]:
```suggestion
與宣告 `useEffect` 本身相同，但它會在所有 DOM 改變後，同步調用。使用它來讀取 DOM layout 並同步重新 render。在瀏覽器執行繪製之前，`useLayoutEffect` 內部的更新將被同步刷新。
```

File: content/docs/hooks-reference.md [hunk 21]
```diff
@@ -406,31 +406,31 @@ function FancyInput(props, ref) {
...
 
-Prefer the standard `useEffect` when possible to avoid blocking visual updates.
+盡可能使用標準的 `useEffect` 來避免阻礙視覺上的更新。
 
-> Tip
+> 提示
```
Comment (neighborhood999) [inline content/docs/hooks-reference.md]:
```suggestion
> 注意
```

File: content/docs/hooks-reference.md [hunk 22]
```diff
@@ -406,31 +406,31 @@ function FancyInput(props, ref) {
...
 >
-> If you're migrating code from a class component, note `useLayoutEffect` fires in the same phase as `componentDidMount` and `componentDidUpdate`. However, **we recommend starting with `useEffect` first** and only trying `useLayoutEffect` if that causes a problem.
+> 如果你是由 class component 遷移代碼，請注意 `useLayoutEffect` 與 `componentDidMount` 和 `componentDidUpdate` 的調用時機是一樣。不過，**我們建議先使用 `useEffect`**，只當它有問題時才嘗試使用 `useLayoutEffect`。
```
Comment (neighborhood999) [inline content/docs/hooks-reference.md]:
```suggestion
> 如果你是從 class component migrate 程式碼，請注意 `useLayoutEffect` 與 `componentDidMount` 和 `componentDidUpdate` 的呼叫時機是一樣。不過，**我們建議先使用 `useEffect`**，只當它有問題時才嘗試使用 `useLayoutEffect`。
```

File: content/docs/hooks-reference.md [hunk 23]
```diff
@@ -406,31 +406,31 @@ function FancyInput(props, ref) {
...
 >
->If you use server rendering, keep in mind that *neither* `useLayoutEffect` nor `useEffect` can run until the JavaScript is downloaded. This is why React warns when a server-rendered component contains `useLayoutEffect`. To fix this, either move that logic to `useEffect` (if it isn't necessary for the first render), or delay showing that component until after the client renders (if the HTML looks broken until `useLayoutEffect` runs).
+>如果你使用伺服器 render，請記著 `useLayoutEffect` 或 `useEffect` *均不會* 運行，直至 JavaScript 完成加載。這是為什麼在伺服器 render 的 component 包含 `useLayoutEffect` 時 React 會發出警告。要解決這問題，把該邏輯搬到 `useEffect` 裏（如果首次 render 不需要該邏輯），或把 component 廷遲到客戶端完成 render 後才出現（如果直到 `useLayoutEffect` 執行前 HTML 都會錯亂的情況下）。
```
Comment (neighborhood999) [inline content/docs/hooks-reference.md]:
```suggestion
>如果你使用伺服器 render，請記住 `useLayoutEffect` 或 `useEffect` *均不會*執行，直到 JavaScript 完成載入。這是為什麼在伺服器 render 的 component 包含 `useLayoutEffect` 時 React 會發出警告。要解決這問題，把該邏輯搬到 `useEffect` 裡（如果首次 render 不需要該邏輯），或把 component 廷遲到客戶端完成 render 後才出現（如果直到 `useLayoutEffect` 執行前 HTML 都會錯亂的情況下）。
```

File: content/docs/hooks-reference.md [hunk 24]
```diff
@@ -406,31 +406,31 @@ function FancyInput(props, ref) {
...
 >
->To exclude a component that needs layout effects from the server-rendered HTML, render it conditionally with `showChild && <Child />` and defer showing it with `useEffect(() => { setShowChild(true); }, [])`. This way, the UI doesn't appear broken before hydration.
+>要在伺服器 render 的 HTML 排除需要 layout effect 的 component，可以利用 `showChild && <Child />` 進行條件 render，並使用 `useEffect(() => { setShowChild(true); }, [])` 來延遲展示。這樣，UI 就不會在完成 render 之前顯示錯亂了。
```
Comment (neighborhood999) [inline content/docs/hooks-reference.md]:
```suggestion
>要在伺服器 render 的 HTML 排除需要 layout effect 的 component，可以利用 `showChild && <Child />` 進行條件 render，並使用 `useEffect(() => { setShowChild(true); }, [])` 來延遲顯示。這樣，UI 就不會在完成 render 之前顯示錯亂了。
```

File: content/docs/hooks-reference.md [hunk 25]
```diff
@@ -406,31 +406,31 @@ function FancyInput(props, ref) {
...
 
 ```js
 useDebugValue(value)
 ```
 
-`useDebugValue` can be used to display a label for custom hooks in React DevTools.
+`useDebugValue` 可以用來在 React DevTools 中顯示自訂義 hook 的標籤。
 
-For example, consider the `useFriendStatus` custom Hook described in ["Building Your Own Hooks"](/docs/hooks-custom.html):
+例如，在[「自定義 hook」](/docs/hooks-custom.html)中提及的 `useFriendStatus` 自定義 hook：
```
Comment (neighborhood999) [inline content/docs/hooks-reference.md]:
```suggestion
例如，在[「打造你的 Hook」](/docs/hooks-custom.html)中提及的 `useFriendStatus` 自定義 Hook：
```

File: content/docs/hooks-reference.md [hunk 26]
```diff
@@ -446,17 +446,17 @@ function useFriendStatus(friendID) {
 }
 ```
 
-> Tip
+> 提示
```
Comment (neighborhood999) [inline content/docs/hooks-reference.md]:
```suggestion
> 注意
```

File: content/docs/hooks-reference.md [hunk 27]
```diff
@@ -446,17 +446,17 @@ function useFriendStatus(friendID) {
...
 >
-> We don't recommend adding debug values to every custom Hook. It's most valuable for custom Hooks that are part of shared libraries.
+> 我們不建議在每個自定義 hook 都加上 debug 值。它在自定義 hook 共享庫中才是最有價值的。
```
Comment (neighborhood999) [inline content/docs/hooks-reference.md]:
```suggestion
> 我們不建議在每個自定義 Hook 都加上 debug 值。它在自定義 Hook 的共享函式庫中才是最有價值的。
```

File: content/docs/hooks-reference.md [hunk 28]
```diff
@@ -446,17 +446,17 @@ function useFriendStatus(friendID) {
...
 
-#### Defer formatting debug values {#defer-formatting-debug-values}
+#### 延遲格式化 debug values {#defer-formatting-debug-values}
```
Comment (neighborhood999) [inline content/docs/hooks-reference.md]:
```suggestion
#### 延遲格式化 Debug Value {#defer-formatting-debug-values}
```

File: content/docs/hooks-reference.md [hunk 29]
```diff
@@ -446,17 +446,17 @@ function useFriendStatus(friendID) {
...
 
-In some cases formatting a value for display might be an expensive operation. It's also unnecessary unless a Hook is actually inspected.
+在某些情況下，格式化一個顯示值可能會是昂貴的操作。而且在直至檢查 hook 前是不必要的。
```
Comment (neighborhood999) [inline content/docs/hooks-reference.md]:
```suggestion
在某些情況下，格式化一個顯示值可能會是昂貴的操作。除非實際檢查 Hook，否則沒有必要。
```

File: content/docs/hooks-reference.md [hunk 30]
```diff
@@ -446,17 +446,17 @@ function useFriendStatus(friendID) {
...
 
-For this reason `useDebugValue` accepts a formatting function as an optional second parameter. This function is only called if the Hooks are inspected. It receives the debug value as a parameter and should return a formatted display value.
+因此，`useDebugValue` 接受一個格式化 function 作為可選的第二個參數。該 function 只有在 hook 被檢查時才會被調用。它接受 debug 值作為參數，然後返回一個已格式化的顯示值。
```
Comment (neighborhood999) [inline content/docs/hooks-reference.md]:
```suggestion
因此，`useDebugValue` 接受一個格式化 function 作為可選的第二個參數。該 function 只有在 Hook 被檢查時才會被呼叫。它接受 debug 值作為參數，然後回傳一個被格式化的顯示值。
```

File: content/docs/hooks-reference.md [hunk 31]
```diff
@@ -446,17 +446,17 @@ function useFriendStatus(friendID) {
...
 
-For example a custom Hook that returned a `Date` value could avoid calling the `toDateString` function unnecessarily by passing the following formatter:
+例如，一個返回 `Date` 值的自定義 hook 可以通過以下的格式化 function 來避免不必要地調用 `toDateString` function：
```
Comment (neighborhood999) [inline content/docs/hooks-reference.md]:
```suggestion
例如，一個回傳 `Date` 值的自定義 Hook 可以通過以下的格式化 function 來避免不必要地呼叫 `toDateString` function：
```

**Comment by ivan-ha** [comment]
thanks @neighborhood999 , all resolved!

**Comment by ivan-ha** [comment]
thanks @neighborhood999, updated again

**Comment by neighborhood999** [comment]
@ivan-ha Thank you! Merged.

**Pull request merged**